column.go

```diff
@@ -785,12 +785,25 @@ func (c *Column) decodeDictionary(header DictionaryPageHeader, page *buffer, siz
 	}
 
 	numValues := int(header.NumValues())
-	values := pageType.NewValues(nil, nil)
+	var (
+		vbuf, obuf  *buffer
+		pageOffsets []uint32
+	)
+	vbuf = buffers.get(pageType.EstimateDecodeSize(numValues, pageData, LookupEncoding(pageEncoding)))
+	defer vbuf.unref()
+
+	if pageType.Kind() == ByteArray && !isDictionaryEncoding(LookupEncoding(pageEncoding)) {
```

Comment:
I basically copied this from the equivalent code when decoding pages. It's necessary because we fall back to plain encoding sometimes which requires offsets. Can remove the `isDictionaryEncoding` check?

Reply:
Yes, at this stage we can drop it since we know we're dealing with a dictionary encoding.